File: Semestrul 4/MPP2/Problema8SC_CSharp/Model/Child.cs
namespace Problema8SC_CSharp.Model;

public class Child : Entity<long>
{
    public long Id
    {
        get => GetId();
        set => SetId(value);
    }
    public String Name { get; set; }
    public String CNP { get; set; }
    
    public Child(long id, String name, String cnp)
    {
        Id = id;
        Name = name;
        CNP = cnp;
    }

    public int getAge()
    {
        return 0;
        //return AgeConverter.getAgeFromCNP(CNP);
    }
    
    public override String ToString()
    {
        return $"Name: {Name}, CNP: {CNP}";
    }
    
    public override bool Equals(object? obj)
    {
        if (obj == null || GetType() != obj.GetType())
        {
            return false;
        }
        
        Child child = (Child)obj;
        return Name.Equals(child.Name) && CNP.Equals(child.CNP);
    }

    protected bool Equals(Child other)
    {
        return Name == other.Name && CNP == other.CNP;
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(Name, CNP);
    }
}

File: Semestrul 4/MPP2/Client-Server_2Limbaje/Problema8SC/Problema8SC/Problema8Model/src/main/java/problema8/model/Child.java
package problema8.model;

import java.util.Objects;

public class Child extends Entity<Long> {

    private String Name;
    private String CNP;

    public Child(Long aLong, String name, String CNP) {
        setId(aLong);
        this.Name = name;
        this.CNP = CNP;
    }

    public String getName() {
        return Name;
    }

    public void setName(String name) {
        this.Name = name;
    }

    public String getCNP() {
        return CNP;
    }

    public void setCNP(String CNP) {
        this.CNP = CNP;
    }

    public Integer getAge() {
        return AgeConverter.getAgeFromCNP(CNP);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Child child = (Child) o;
        return Objects.equals(Name, child.Name) && Objects.equals(CNP, child.CNP);
    }

    @Override
    public int hashCode() {
        return Objects.hash(Name, CNP);
    }

    @Override
    public String toString() {
        return "Child{" +
                "name='" + Name + '\'' +
                ", CNP='" + CNP + '\'' +
                '}';
    }
}
